packages/stark-ui/src/modules/app-sidebar/components/app-sidebar.component.ts

```diff
@@ -0,0 +1,123 @@
+import { Component, HostBinding, Inject, Input, OnDestroy, OnInit, ViewChild, ViewEncapsulation } from "@angular/core";
+import { from, Subscription } from "rxjs";
+import { MatSidenav, MatSidenavContainer, MatDrawerToggleResult } from "@angular/material/sidenav";
+import { STARK_LOGGING_SERVICE, StarkLoggingService } from "@nationalbankbelgium/stark-core";
+import { StarkAppSidebarOpenEvent } from "../services/app-sidebar-open-event.intf";
+import { StarkAppSidebarService } from "../services/app-sidebar.service";
+
+/**
+ * Name of the component
+ */
+const componentName: string = "stark-app-sidebar";
+
+/**
+ * Component to display the application's sidebar
+ * Only 2 sidebars are allowed: https://github.com/angular/material2/issues/1514
+ */
+@Component({
+	selector: "stark-app-sidebar",
+	templateUrl: "./app-sidebar.component.html",
+	encapsulation: ViewEncapsulation.None
+})
+export class StarkAppSidebarComponent implements OnDestroy, OnInit {
+	/**
+	 * Adds class="stark-app-sidebar" attribute on the host component
+	 */
+	@HostBinding("class")
+	public class: string = componentName;
+
+	/**
+	 * Mode for the left sidebar: either the menu is shown or the regular sidebar
+	 */
+	@Input()
+	public sidenavLeftMode: "regular" | "menu";
```

Comment:
Can you create a custom type for these? That way you can re-use them in the  StarkAppSidebarOpenEvent interface ;)